packages/client/components/ActivityLibrary/ActivityDetails.tsx

```diff
@@ -205,6 +205,10 @@ const ActivityDetails = (props: Props) => {
 
   const [isEditing, setIsEditing] = useState(false)
 
+  useEffect(() => {
+    setIsEditing(!!history.location.state?.edit)
+  }, [history.location.state?.edit, setIsEditing])
```

Comment:
-1 no need to use `useEffect` here, you can init the state like 
```
  const [isEditing, setIsEditing] = useState(() => !!history.location.state?.edit)
```

unless I'm missing some case when we actually want to observe location state

Reply:
This is what I tried originally, but if we clone a template, React will reuse this component, keeping the previous template's edit state, so we need to set this manually

Reply:
ah interesting! let's go with that solution, I'll investigate it a bit more in the meantime